src/recipe.h

```diff
@@ -282,9 +283,9 @@ class recipe
         std::vector<std::pair<std::string, int>> bp_excludes;
 
         /** Blueprint requirements to be checked in unit test */
-        bool has_blueprint_needs = false;
+        bool bp_autocalc = false;
         bool check_blueprint_needs = false;
-        build_reqs blueprint_reqs;
+        cata::optional<build_reqs> blueprint_reqs_;
```

Comment:
Using `std::unique_ptr` here could save a few bytes of memory after loading is completed.

Reply:
I thought it wouldn't be a big saving, but it's actually >100 bytes, so I agree.  Done.